src/accelerate/state.py

```diff
@@ -797,6 +809,21 @@ def default_device(self) -> torch.device:
         else:
             return torch.device("cpu")
 
+    def __getattr__(self, name: str):
+        try:
+            # First try the normal way
+            return object.__getattribute__(self, name)
+        except AttributeError as e:
+            if name in self._known_attrs:
+                # If not found, give a more contextualized answer
+                raise AttributeError(
+                    f"`PartialState` object has no attribute `{name}`. "
+                    "This happens if `PartialState._reset_state()` was called and "
+                    "an `Accelerator` or `PartialState` was not reinitialized."
```

Comment:
Do we need "an" here?

Reply:
For grammar I'd say yes, but no option one way or the other